### DISCUSSION TITLE
hydraulic-head gradient (initial condition)

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#18014] [Q&A Modules: General] [answered]
Hi guys,

I am trying to set a hydraulic-head gradient on 3D geometry, is there any simple method to work around this.

I tried to set the side boundary with different gradients but that doesn't give me the results I want (I want it to follow the red line as attached).

Thank you guys.

@WilkAndy I think you are the expert on this topic, you might be able to quickly help, thanks!

Kind regards,
Traiwit

```
[ICs]
  [./porepressure]
    type = FunctionIC
    variable = porepressure
    function = ini_pp
  [../]
[]

[BCs]
     [./pp_z2]
       type = DirichletBC
       variable = porepressure
       boundary = 'back'
       value = -9810000
     [../]
     [./pp_z1]
       type = DirichletBC
       variable = porepressure
       boundary = 'front'
       value = 0
     [../]
     [./water_grad_x1]
       type = FunctionDirichletBC
       variable = porepressure
       boundary = 'bottom'
       function = watergradx1
     [../]
     [./water_grad_x2]
       type = FunctionDirichletBC
       variable = porepressure
       boundary = 'top'
       function = watergradx2
     [../]
[]

[Functions]
  [./ini_pp]
    type = PiecewiseLinear
    axis = z
      x = '0 900 1000'
      y = '-9810000 0 0'
  [../]
  [./watergradx1]
    type = PiecewiseLinear
    axis = z
    x = '0 900 1000'
    y = '-9810000 0 0'
  [../]
  [./watergradx2]
    type = PiecewiseLinear
    axis = z
    x = '0 700 1000'
    y = '-9810000 0 0'
  [../]
[]
```

![2021-06-07_11-44](https://user-images.githubusercontent.com/79562439/120948900-938d7b00-c786-11eb-9b2c-30f34530e604.png)

**Comment by Traiwit** [comment]
okay guys, think about it again, this is physically impossible, since the fluid level probs should be controlled by Pc. Moreover, the fluid contact in the same hydraulic system should be flat.

**Comment by WilkAndy** [comment, marked as answer]
In steady-state, given the same BCs on all the vertical sides of the model (eg, hydrostatic) this head configuration is impossible, as you note.  However , these types of hydraulic gradients are always observed in nature.  This is usually because of some sort of recharge that is happening at y=-big (in your model) - often in some nearby mountain range - and discharge that is happening at y=big (in your model) - often in a valley or floodplain or ocean, etc - along with an aquitard "cap" in the model domain that prevents vertical movement of water to the topography.

To get this initial condition in a model can be difficult or hard, depending on how you approach the problem.  You can take the hard route by extending your model domain so it includes the recharge and discharge points, and running a sequence of models with different recharge/discharge and hydraulic properties until your model matches observations.  This is possibly the "best" type of modelling approach since it most strongly suggests you've got a model that reflects reality.  However, doing the calibration to observation is rather time consuming, and often when you extend the model you're encountered with a bunch of unknowns, which makes your model "weaker" since you have to somehow prescribe hydraulic properties, lithologies, recharge zones, etc, that you don't know anything much about.

Instead, you often simply fix the BCs such that the steady-state matches your observations, without worrying too much about why or how those boundary conditions arise.  Of course, this would be inappropriate if the observations have been influenced by recent water abstractions (in this case, the BCs might be simple hydrostatic, but the ICs would be what you describe).  In MOOSE, you could use`Function`s to define these BCs and ICs.  I often use a `ParsedFuntion`, and always try to use only 1 or a small number of such Functions (e.g. use the same function in all BCs and ICs).  Throughout the aquifers you can assume hydrostatic, unless it disagrees with observations.  These hydrostatic lines "jump" as you go through the aquitards.   I hope you understand what i mean: as you move vertically the porepressure (or head) looks like a lightning bolt: :zap: .

a

**Reply by Traiwit** [reply]
Noted, thank you @WilkAndy,

May I have a quick question about the excavation with hydro.

How do I let the pore-pressure of the excavated block = 0, it seems that I cannot control it via [Materials] block like Young modulus.

From your coal-mining problem, I see you control it with the Sink term (but that is at the boundary)
I am not sure if it's possible to do the same in the middle of the geometry.

Kind regards,
Traiwit

```
  [./roof_porepressure]
    type = PorousFlowPiecewiseLinearSink
    variable = porepressure
    pt_vals = '-1E3 1E3'
    multipliers = '-1 1'
    fluid_phase = 0
    flux_function = roof_conductance
    boundary = roof
  [../]

  [./roof_conductance]
    type = ParsedFunction
    vars = 'end_t ymin ymax   maxval minval'
    vals = '0.5   0    1000.0 1E7      0'
    value = 'if(y<ymin+(ymax-ymin)*min(t/end_t,1),maxval,minval)'
  [../]
```

**Reply by WilkAndy** [reply]
Why do you care about the porepressure in an excavated block, since it has been removed from your model?

**Reply by Traiwit** [reply]
true that's a very valid point, so basically I should just set porepressure BC around the excavated block to be 0 to model the movement of the fluid after excavation.

Thanks again @WilkAndy

**Reply by WilkAndy** [reply]
Yep, a `DirichletBC` or `FunctionDirichletBC` will do nicely, i think.   (The advantage of the `PorousFlowPiecewiseLinearSink` is that you have a bit more control on how much water is coming into the excavation, but `DirichletBC` will be fine for you.)

**Reply by Traiwit** [reply]
Perfect got it, thanks again @WilkAndy 

```

  [add_side_sets_block6]
    type = SideSetsAroundSubdomainGenerator
    block = 6
    input = add_side_sets
    normals = '0  0  1
               0  1  0
               1  0  0
               0  0 -1
               0 -1  0
              -1  0  0'
    fixed_normal = true
    new_boundary = 'front6 top6 right6 back6 bottom6 left6'
    variance = 0.5
  []

  [./water_grad_block6]
            type = FunctionDirichletBC
            variable = porepressure
            boundary = 'front6 top6 right6 back6 bottom6 left6'
            function = pp_block6
  [../]

[./pp_block6]
    type = ParsedFunction
  value = 'if(t>2,0,-9.81*1000*(1000-z))'
  [../]
```
![2021-06-07_15-42](https://user-images.githubusercontent.com/79562439/120964824-3eae2c80-c7a7-11eb-8bd7-fb1fcc46b6c2.png)
